File: backend/src/main/java/com/godeltech/mastery/backend/repository/CarRepository.java
package com.godeltech.mastery.backend.repository;

import com.godeltech.mastery.backend.domain.entity.Car;
import com.godeltech.mastery.backend.domain.entity.Client;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

public interface CarRepository extends JpaRepository<Car, Long>, JpaSpecificationExecutor<Car> {

  List<Car> getAllByClient(Client client);
}

File: backend/src/main/kotlin/com/godeltech/mastery/backend/repository/CarRepository.kt
package com.godeltech.mastery.backend.repository

import com.godeltech.mastery.backend.domain.entity.Car
import com.godeltech.mastery.backend.domain.entity.Client
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor

interface CarRepository : JpaRepository<Car, Long>, JpaSpecificationExecutor<Car> {

    fun getAllByClient(client: Client): List<Car>
}